File: java/problem459.java
class Solution {
    public boolean repeatedSubstringPattern(String s) {
        int size=s.length();
        boolean tag=false;
        for(int i=1;i<size;i++){
            tag=false;
            if(size%i==0){//the length of substing must be the factor of total length
                tag=true;
                for(int j=0;j<size/i-1;j++){//check substring(0--->i)
                    for(int k=0;k<i;k++){
                        if(s.charAt(k)!=s.charAt(k+(j+1)*i)){
                            tag=false;
                            break;
                        }
                    }
                    if(!tag){
                        break;
                    }
                }
            }
            if(tag){
                break;
            }
        }
        return tag;
        
    }
}

File: python/problem459.py
# from https://leetcode.com/problems/repeated-substring-pattern/discuss/94334/Easy-python-solution-with-explaination
class Solution:
    def repeatedSubstringPattern(self, s):
        """
        :type s: str
        :rtype: bool
        """
        if not s:
            return False
            
        ss = (s + s)[1:-1]
        return ss.find(s) != -1
